mautic/mautic
# Segment query optimization

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [N]
| New feature/enhancement? (use the a.x branch)      | [Y]
| Deprecations?                          | [N]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [Y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


This will increase the performance of segment queries as it will add lead limiters to inner query and lower rows scanned.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create segment with filters and test every filter is working fine.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11283"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
